### PR TITLE
Convert source code to typescript

### PR DESCRIPTION
**Pull request by elpoelma**
This PR converts the source code in this repository to typescript.
Additionally, updated ESLint and tsconfig.json files are added.

Note: some dependencies of this application do not have type declarations yet. In these cases a `@ts-expect-error` comment is added. When type declarations are added to these dependencies, `ts-expect-error` will yield errors.

The official built-in ember-data types are not yet released at the moment. Instead, this PR makes uses of the following DefinitelyTyped packages:
-    `@types/ember-data`
-    `@types/ember-data__adapter`
-    `@types/ember-data__model`
-    `@types/ember-data__serializer`
-    `@types/ember-data__store`

These types were last updated for ember 4.4. These can still contain some erroneous types.

### CODE REVIEW

File: app/components/mu-login.hbs [hunk 1]
```diff
@@ -4,19 +4,19 @@
...
   <div>
     <AuLabel for="password">
       {{t "utility.password"}}&nbsp;
     </AuLabel>
     <AuInput
-      type="password"
+      @type="password"
       id="password"
-      {{on "input" (fn this.setPassword this.password)}}
+      @value={{this.password}}
```
Comment (x-m-el) [inline app/components/mu-login.hbs:19]: small note: I think you don't want to change this to only a @value hook, as this is a double-bind which might disappear (or get deprecated) in the future: https://github.com/appuniversum/ember-appuniversum/issues/250

so instead `{{on 'input' this.setPassword}}` and in class: `setPassword(event) {this.password = event.target.value;}`

Comment (elpoelma) [reply app/components/mu-login.hbs:19]: Ok makes sense, I'll make the required changes.

**Review by x-m-el — APPROVED** [review]
Not something to really easily review, without spending a lot of time.
As far as I can tell:
No (new) bugs as far as I can find.
Code looks sensible, at least with a "quick" look.

**Comment by elpoelma** [comment]
In relation to https://github.com/emberjs/data/issues/8599, we could maybe take the approach mentioned in this issue (casting to `unknown` and the to an `array` when resolving hasMany relationships.

**Pull request merged**